### DISCUSSION TITLE
Error: [storage/unknown] An unknown error has occurred. iOS firebase

### DISCUSSION THREAD

**Discussion by VikashSRSpritle** [invertase/react-native-firebase#6399] [Q&A]
I'm trying to post some csv files to Firebase by uploading them to Firestore, downloading the urls, and then doing so, but I keep receiving this issue. Ninety percent of the time, it works, but occasionally it displays this error.

**Error: [storage/unknown] An unknown error has occurred.**

```
const uploadToStorage = async (file, index) => {
    await storage()
      .ref(`SensorData/${userName}/session_${sessionCount}/${index}.csv`)
      .putFile(file);
    const url = await storage()
      .ref(`SensorData/${userName}/session_${sessionCount}/${index}.csv`)
      .getDownloadURL();
    urls[index] = url;
    return urls;
  };

const uploadToFirebase = async (files) => {
    urls = [];
    Promise.all(
      // Array of "Promises"
      files.map((file, index) => uploadToStorage(file, index))
    )
      .then((url) => {
        postReportsToFirebase(urls);
      })
      .catch((error) => {
        console.log(error);
      });
  };
```
Version:

"react-native": "0.64.1",
"@react-native-firebase/app": "^12.7.3",
"@react-native-firebase/auth": "^12.7.3",
"@react-native-firebase/firestore": "^12.7.3",
"@react-native-firebase/functions": "^12.7.3",
"@react-native-firebase/storage": "^12.7.3",*

**Comment by mikehardy** [comment]
Hi there! You are likely going to want this commit in order to see what happened: https://github.com/invertase/react-native-firebase/commit/6cd53eaca16ef52c52a28a7b209a7c8313fef08b

But it is only available with the v15 versions and higher of react-native-firebase and I do understand using those can be quite problematic since they imply adding `use_frameworks!` to your Podfile and that is very hard to do in the react-native ecosystem currently (though I'm working on that - it still needs work)

You might try to port that change into your current app via use of patch-package (https://github.com/ds300/patch-package/) and then see if you can get a better idea of what the exact error is, it might point to the problem directly

If it does not, the next step I think is to add logging directly into the storage objective-c code in react-native-firebase/storage inside node_modules in order to see what is happening with the API call / error return, and watch the Xcode console after launching the app

Hope this helps you figure it out

**Comment by ics-ankit** [comment]
same issue i am uploading the image .jpg from similator but not upload and give that error 

+1  same issue any solution
